### PR TITLE
Additional state check.

### PR DESCRIPTION
**Pull request by cheme**
This PR adds an additional check on state root at the end of client `block_execution` method.
This check is redundant with the one in runtime finalize block, but has a really low cost.
We could also consider that it is something that should be done in runtime only, and replace this test by an assertion or debug assertion, but if the test fail there is high chance that the new root is not saved and data will be lost.
There is also a slight change in test runtime code : a modification post root checking was in code (which this test detect).

### CODE REVIEW

**Comment by cheme** [comment]
Ok that is not acting well with grandpa, I will investigate further, closing it for now.

**Comment by cheme** [comment]
fixed the granpa test (test runtime was modifying root after its check see https://github.com/paritytech/substrate/pull/2915/commits/90bc2614ba100913ee57a7e045ea55229b523df6.

File: core/test-runtime/src/system.rs [hunk 1]
```diff
@@ -124,7 +124,7 @@ fn execute_block_with_state_root_handler(
 	if let Some(storage_changes_root) = storage_changes_root(header.parent_hash.into()) {
 		digest.push(generic::DigestItem::ChangesTrieRoot(storage_changes_root.into()));
 	}
-	if let Some(new_authorities) = <NewAuthorities>::take() {
+	if let Some(new_authorities) = <NewAuthorities>::get() {
```
Comment (bkchr) [inline core/test-runtime/src/system.rs]: Hmm, but then this needs to be reset somewhere else?

Comment (cheme) [reply core/test-runtime/src/system.rs]: yes you're right, we want to get both root calculation to not include the set value, there is probably more to it, will check later.

Comment (cheme) [reply core/test-runtime/src/system.rs]: simply removing the value before root calculation seems fine.

**Comment by gavofyork** [comment]
best to bump runtime.

File: core/sr-api-macros/tests/trybuild.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#[test]
+//#[test]
```
Comment (bkchr) [inline core/sr-api-macros/tests/trybuild.rs]:
```suggestion
#[test]
```

**Comment by cheme** [comment]
Should be (commented test was related to local failing test due rust version I think, still waiting for the CI).

**Pull request merged**